File: backend/api-gateway/src/main/java/ru/animaltracker/apigateway/repository/UserRepository.java
package ru.animaltracker.apigateway.repository;

import reactor.core.publisher.Mono;
import ru.animaltracker.apigateway.entity.User;
import org.springframework.data.r2dbc.repository.R2dbcRepository;

public interface UserRepository extends R2dbcRepository<User, Long> {
    Mono<User> findByEmail(String email);
    Mono<User> findByUsername(String username);
}

File: backend/user-service/src/main/kotlin/ru/animaltracker/userservice/repository/UserRepository.kt
package ru.animaltracker.userservice.repository

import org.springframework.data.jpa.repository.JpaRepository
import ru.animaltracker.userservice.entity.User

interface UserRepository : JpaRepository<User, Long> {
    fun findByEmail(email: String): User?
    fun findByUsername(username: String): User?
}